bokehjs/src/lib/embed/index.ts

```diff
@@ -1,38 +1,38 @@
 import {Document, DocJson} from "../document"
+import {ID} from "../core/types"
 import {logger} from "../core/logging"
 import {unescape, uuid4} from "../core/util/string"
 import {entries} from "core/util/object"
 import {isString} from "../core/util/types"
 import {defer} from "core/util/defer"
 import {ViewManager} from "core/view"
 
-import {DocsJson, RenderItem} from "./json"
+import {DocsJson, RenderItem, Roots} from "./json"
 import {add_document_standalone} from "./standalone"
 import {add_document_from_session, _get_ws_url} from "./server"
 import {BOKEH_ROOT, _resolve_element, _resolve_root_elements} from "./dom"
 
-export {DocsJson, RenderItem} from "./json"
+export {DocsJson, RenderItem, Roots} from "./json"
 export {add_document_standalone, index} from "./standalone"
 export {add_document_from_session} from "./server"
 export {embed_items_notebook, kernels} from "./notebook"
 export {BOKEH_ROOT} from "./dom"
 
-export type JsonItem = {doc: DocJson, root_id: string, target_id: string}
-type Roots = {[index: string]: string}
+export type JsonItem = {doc: DocJson, root_id: ID, target_id: ID}
 
-export async function embed_item(item: JsonItem, target_id?: string): Promise<ViewManager> {
+export async function embed_item(item: JsonItem, target?: ID | HTMLElement): Promise<ViewManager> {
```

Comment:
Perhaps we should have allowed CSS selectors instead of only IDs in all such cases. I would like to change that, but it would be quite a fragile change and hard to update existing codebases.